### PR TITLE
Add support for writing Day2 API tests

### PR DESCRIPTION
**Pull request by mchernik**
- Reuse existing TF configuration
  - Always return fresh TF state
  - Clear nodes cache
  - Fetch reuested number of Day2 nodes from environment

### CODE REVIEW

File: discovery-infra/test_infra/tools/terraform_utils.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def change_variables(self, variables, refresh=True):
         self.apply(refresh=refresh)
 
     def get_state(self):
+        self.tf.read_state_file("terraform.tfstate")
```
Comment (omertuc) [inline discovery-infra/test_infra/tools/terraform_utils.py:39]: Add `STATE_FILE` constant to class scope and reuse it both here and in `__init__`

Comment (omertuc) [reply discovery-infra/test_infra/tools/terraform_utils.py:39]: (just like `VAR_FILE`)

File: discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 2]
```diff
@@ -15,14 +15,15 @@ class TerraformController(LibvirtController):
...
+        self.cluster_suffix = kwargs.get('cluster_suffix', self._get_random_name())
         self.cluster_name = kwargs.get('cluster_name', f'{consts.CLUSTER_PREFIX}' + "-" + self.cluster_suffix)
         self.network_name = kwargs.get('network_name', consts.TEST_NETWORK) + self.cluster_suffix
         self.network_conf = kwargs.get('net_asset')
         self.cluster_domain = kwargs.get('base_domain', "redhat.com")
         self.ipv6 = kwargs.get('ipv6')
         self.params = self._terraform_params(**kwargs)
-        self.tf_folder = self._create_tf_folder()
+        tf_folder = kwargs.get('tf_folder')
+        self.tf_folder = tf_folder if tf_folder else self._create_tf_folder()
```
[the remark this answers is not in the document]
Comment (mchernik) [reply discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py:26]: Yes. The default parameter to `get`  ( `_create_tf_folder()` method) is evaluated before calling `get`, so the TF folder is always recreated with this approach.

**Pull request merged**